File: Array/find_all_missing.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        # count from 1 to n
        # each iteration find n exist ?
        # if not exist push back to list
        n = len(nums)
        nums_set = set(nums)
        ans = []
        for i in range(1, n+1):
            if i not in nums_set:
                ans.append(i)

        return ans

File: Array/find_all_missing.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        // loop thr to map between nums and index
        int tmp_index = 0;
        for (int i = 0; i < nums.size(); i++) {
            tmp_index = abs(nums[i]) - 1;
            if (nums[tmp_index] > 0) {
                nums[tmp_index] = -nums[tmp_index];
            }
        }

        vector<int> ans;
        // find non-neg and append into ans
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] > 0) {
                ans.push_back(i + 1);
            }
        }

        return ans;
    }
};